inlong-sort/sort-connectors/filesystem/src/main/java/org/apache/inlong/sort/filesystem/stream/AbstractStreamingWriter.java

```diff
@@ -129,7 +136,20 @@ public void open() throws Exception {
      * Commit up to this checkpoint id.
      */
     protected void commitUpToCheckpoint(long checkpointId) throws Exception {
-        helper.commitUpToCheckpoint(checkpointId);
+        try {
+            helper.commitUpToCheckpoint(checkpointId);
+            if (sinkMetricData != null) {
+                sinkMetricData.invoke(rowSize, dataSize);
+            }
+        } catch (Exception e) {
```

Comment:
It is recommended to throw an exception first to ensure that data is not lost.

Reply:
> It is recommended to throw an exception first to ensure that data is not lost.

@yunqingmoswu dirty data metric can be reportered if throw exception